blocks/mosaic-reveal/mosaic-reveal.js

```diff
@@ -0,0 +1,75 @@
+import { h3, div, domEl } from '../../scripts/dom-helpers.js';
+
+export default function decorate(block) {
+  block.classList.add('calcite-mode-dark');
+
+  [...block.children].forEach((child) => {
+    const titleText = child.querySelector('h3').textContent;
+
+    child.children[0].appendChild(h3({ class: 'title' }, titleText));
+
+    child.appendChild(domEl('calcite-icon', { icon: 'expand', scale: 's' }));
+  });
+
+  const contents = [...block.children].map((child) => child.children[1]);
+  contents.forEach((content) => {
+    content.classList.add('mosaic-reveal-content');
+    content.setAttribute('aria-hidden', 'true');
+
+    const buttonContainer = content.querySelector('p.button-container > a');
+    const linkText = buttonContainer.textContent;
+    const url = buttonContainer.href;
+
+    const link = domEl('calcite-link', {
+      href: url,
+      'icon-end': 'arrowRight',
+    }, linkText);
+
+    buttonContainer.parentElement.appendChild(link);
+    buttonContainer.parentElement.removeChild(buttonContainer);
+  });
+
+  const revealContent = div({ class: 'mosaic-reveal-content', 'aria-hidden': 'true' });
+  revealContent.addEventListener('click', () => {
+    revealContent.setAttribute('aria-hidden', 'true');
+  });
+
+  window.addEventListener('resize', () => {
```

Comment:
you might want to use window.matchMedia instead:
https://github.com/hlxsites/moleculardevices/blob/15619942358776750c1f4280730f98b6e4e6cd37/blocks/citation-list/citation-list.js#L71
https://developer.mozilla.org/en-US/docs/Web/API/Window/matchMedia